### PR TITLE
add local-up-master for running a controller manager without a kubelet

### PR DESCRIPTION
**Pull request by juanvallejo**
Picks up https://github.com/openshift/origin/pull/21125

### CODE REVIEW

**Comment by juanvallejo** [comment]
Addressing some comments from previous PR:

cc @soltysh 
> It looks like we can't rely on glide.lock either since it'll get you etcd-io-etcd-v3.2.16-1-g135cf9b.tar.gz and that won't pass your initial check for 3.2.24.

I was able to get this to run with etcd v3.2.16. I have changed the version requirement accordingly, in order to continue relying on Godeps.json for this

File: hack/local-up-master/lib.sh [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+MAX_TIME_FOR_URL_API_SERVER=${MAX_TIME_FOR_URL_API_SERVER:-1}
+
+#source "$(dirname "${BASH_SOURCE}")/../lib/init.sh"
+KUBE_ROOT=OS_ROOT
+source "${OS_ROOT}/hack/local-up-master/logging.sh"
+source "${OS_ROOT}/hack/local-up-master/util.sh"
+source "${OS_ROOT}/hack/local-up-master/etcd.sh"
+
+
+CONFIG=$(pwd)/openshift.local.masterup
```
Comment (juanvallejo) [inline hack/local-up-master/lib.sh]: @soltysh 
> Why not using /tmp ?

I think it makes sense to set this to use /tmp when running it from test-cmd. But when running as standalone apiserver, it follows same behavior of `oc cluster up` by placing config directory in current working path

File: hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml [hunk 2]
```diff
@@ -0,0 +1,159 @@
+apiVersion: v1
```
Comment (juanvallejo) [inline hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml]: @soltysh 
> Maybe a script to simplify this?

Not sure what you mean here, should we render these files instead?

**Comment by juanvallejo** [comment]
cc @deads2k @soltysh

**Comment by deads2k** [comment]
@juanvallejo is that a green test-cmd??  Is it broken or commented out some how?

**Comment by deads2k** [comment]
oh, wrong pull

**Comment by juanvallejo** [comment]
:) Hopefully it will be a green test-cmd in the right PR soon

/retest

**Pull request merged**